frontend/src/metabase/common/hooks/use-toast/use-toast.ts

```diff
@@ -0,0 +1,14 @@
+import { useDispatch } from "metabase/lib/redux";
+import { addUndo, dismissUndo } from "metabase/redux/undo";
+import type { Undo } from "metabase-types/store/undo";
+
+type ToastArgs = Omit<Undo, "id" | "timeoutId">;
+
+// A handy convenience hook for adding toast/undo notifications
+export const useToast = () => {
+  const dispatch = useDispatch();
+  const sendToast = (toastInfo: ToastArgs) => dispatch(addUndo(toastInfo));
+  const dismissToast = (toastId: number) =>
+    dispatch(dismissUndo({ undoId: toastId }));
```

Comment:
Please wrap both of these functions in `useCallback` so that they are memoized and play nicely with being put in hook dependency arrays in the components they are used in.